### PR TITLE
[orientdb] added test cases

### PR DESCRIPTION
**Pull request by kruthar**
Added basic operational test cases for the orientdb binding. Should give a little assurance to future cleanup work.

### CODE REVIEW

**Comment by kruthar** [comment]
So, this was a weird failure, seem that whatever location clojars was could not be reached.

I reran the build and things are green.

File: orientdb/src/test/java/com/yahoo/ycsb/db/OrientDBClientTest.java [hunk 1]
```diff
@@ -0,0 +1,228 @@
+package com.yahoo.ycsb.db;
```
Comment (busbey) [inline orientdb/src/test/java/com/yahoo/ycsb/db/OrientDBClientTest.java]: File needs a license header.

File: orientdb/src/test/java/com/yahoo/ycsb/db/OrientDBClientTest.java [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+      Properties p = new Properties();
+      // TODO: Extract the property names into final variables in OrientDBClient
+      p.setProperty("orientdb.url", TEST_DB_URL);
+
+      orientDBClient.setProperties(p);
+      orientDBClient.init();
+      orientDBConnection = new ODatabaseDocumentTx(TEST_DB_URL).open("admin","admin");
+      orientDBDictionary = orientDBConnection.getMetadata().getIndexManager().getDictionary();
+    } catch (DBException e) {
```
Comment (busbey) [inline orientdb/src/test/java/com/yahoo/ycsb/db/OrientDBClientTest.java]: If you just let this exception propogate out of the method, junit will take care of logging the details.

File: orientdb/src/test/java/com/yahoo/ycsb/db/OrientDBClientTest.java [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    if (orientDBConnection != null) {
+      orientDBConnection.close();
+    }
+
+    try {
+      if (orientDBClient != null) {
+        orientDBClient.cleanup();
+      }
+    } catch (DBException e) {
+      e.printStackTrace();
```
Comment (busbey) [inline orientdb/src/test/java/com/yahoo/ycsb/db/OrientDBClientTest.java]: Same comment as on the setup method re: letting the exception propogate.

File: orientdb/src/test/java/com/yahoo/ycsb/db/OrientDBClientTest.java (hunk 1 above)
Comment (kruthar) [inline orientdb/src/test/java/com/yahoo/ycsb/db/OrientDBClientTest.java]: Indeed it does. Added.

**Comment by kruthar** [comment]
@busbey - thanks for the tip, exceptions now propogating

File: orientdb/src/test/java/com/yahoo/ycsb/db/OrientDBClientTest.java [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+    }
+  }
+
+  /*
+      This is a copy of buildDeterministicValue() from core:com.yahoo.ycsb.workloads.CoreWorkload.java.
+      That method is neither public nor static so we need a copy.
+   */
+  private String buildDeterministicValue(String key, String fieldkey) {
```
Comment (busbey) [inline orientdb/src/test/java/com/yahoo/ycsb/db/OrientDBClientTest.java]: Any thoughts on this copying vs using something like powermock to access the method vs making the original visible for testing?

File: orientdb/src/test/java/com/yahoo/ycsb/db/OrientDBClientTest.java [hunk 5]
```diff
@@ -0,0 +1,236 @@
+/**
+ * Copyright (c) 2015 YCSB contributors. All rights reserved.
```
Comment (busbey) [inline orientdb/src/test/java/com/yahoo/ycsb/db/OrientDBClientTest.java]: 2016?

File: orientdb/src/test/java/com/yahoo/ycsb/db/OrientDBClientTest.java (hunk 4 above)
Comment (busbey) [inline orientdb/src/test/java/com/yahoo/ycsb/db/OrientDBClientTest.java]: Thinking about this more, I agree that copying makes the most sense right now. It probably points to a need to come up with a common strategy for these kinds of tests, but that's not an issue for this PR.

File: orientdb/src/test/java/com/yahoo/ycsb/db/OrientDBClientTest.java (hunk 5 above)
Comment (kruthar) [inline orientdb/src/test/java/com/yahoo/ycsb/db/OrientDBClientTest.java]: I actually wrote this in 2015, but the PR came in 2016, does one matter over the other?

Comment (busbey) [inline orientdb/src/test/java/com/yahoo/ycsb/db/OrientDBClientTest.java]: Wrote in 2015 means 2015 is correct. Just wanted to double check.

**Comment by busbey** [comment]
+1

File: orientdb/src/test/java/com/yahoo/ycsb/db/OrientDBClientTest.java (hunk 4 above)
Comment (kruthar) [inline orientdb/src/test/java/com/yahoo/ycsb/db/OrientDBClientTest.java]: Agreed on your last point, needs deeper thought on consolidated testing strategy in the future.

**Comment by kruthar** [comment]
Thanks @busbey

**Pull request merged**